test/k23si/QueryTest.cpp

```diff
@@ -120,14 +121,15 @@ class QueryTest {
 
 seastar::future<> doQuery(const k2::String& start, const k2::String& end, int32_t limit, bool reverse, 
                           uint32_t expectedRecords, uint32_t expectedPaginations, 
-                          k2::Status expectedStatus=k2::dto::K23SIStatus::OK) {
+                          k2::Status expectedStatus=k2::dto::K23SIStatus::OK,
+                          std::vector<k2::String> projection=std::vector<k2::String>()) {
```

Comment:
I think you should change this function so it returns the result set to the caller, then the tests for projection can test that the correct data is returned. The already existing tests can just ignore the result.